proxies/ruby/app.rb

```diff
@@ -19,49 +22,89 @@
 
 data_store = {
   clients: {},
-  commandResults: {}
+  command_results: {}
 }
 
 post '/client' do
   request.body.rewind
-  data = JSON.parse request.body.read
+  body = JSON.parse request.body.read
 
   client_id, sdk_key, wait_for_initialization, options =
-    data.values_at('clientId', 'sdkKey', 'waitForInitialization', 'options')
+    body.values_at('clientId', 'sdkKey', 'waitForInitialization', 'options')
 
   if client_id.nil?
     status 400
     return { message: 'Invalid request: missing clientId' }.to_json
   end
 
   begin
-    options = DevCycle::DVCOptions.new(
-      config_cdn_uri: options.fetch('configCDNURI', 'https://config-cdn.devcycle.com'),
-      events_api_uri: options.fetch('eventsAPIURI', 'https://events.devcycle.com'),
+    dvc_options = DevCycle::DVCOptions.new(
+      config_cdn_uri: options.fetch('configCDNURI', ''),
+      events_api_uri: options.fetch('eventsAPIURI', ''),
       config_polling_interval_ms: options.fetch('configPollingIntervalMS', 10_000),
       event_flush_interval_ms: options.fetch('eventFlushIntervalMS', 1000)
     )
 
-    async_error = nil
-
-    if wait_for_initialization
-      begin
-        data_store[:clients][client_id] = DevCycle::DVCClient.new(sdk_key, options, true)
-      rescue StandardError => e
-        async_error = e
-      end
-    else
-      data_store[:clients][client_id] = DevCycle::DVCClient.new(sdk_key, options, false)
-    end
+    data_store[:clients][client_id] = DevCycle::DVCClient.new(sdk_key, dvc_options, wait_for_initialization)
 
     status 201
     headers 'Location' => "/client/#{client_id}"
 
-    return { asyncError: async_error }.to_json if async_error
-
     { message: 'success' }.to_json
   rescue StandardError => e
     status 200
-    { error: e.message }.to_json
+    { exception: e.message }.to_json
```

Comment:
do we always send initialization errors as "exception"? is that why this change?

Reply:
for nodejs we have code that sends `exception` or `asyncError` based on where the error occurs but the `clientInitialized` never actually throws so we decided for Go and Dotnet (Ruby now too) to never report `asyncErorrors`